tests/reference/asr-test_annassign_type_mismatch-7dac7be.stderr

```diff
@@ -2,4 +2,4 @@ semantic error: Type mismatch in annotation-assignment, the types must be compat
  --> tests/errors/test_annassign_type_mismatch.py:4:5
   |
 4 |     x: i32[4] = [1, 2, 3, 4]
-  |     ^           ^^^^^^^^^^^^ type mismatch ('i32[4]' and 'list[i32]')
+  |     ^           ^^^^^^^^^^^^ type mismatch ('i32[0,4]' and 'list[i32]')
```

Comment:
This is a bug. The previous error message was correct.

Reply:
I agree. I was trying to also include the `start` index of the dimension in the error message as we now check for dimensions when we use `empty()` (previously we used to ignore/skip the call to empty including skipping the dimensions and the dtype). We now check both `start` and `length` of the dimension to match with that of the dimensions of the variable to which the value of `empty()` is being assigned to. The above mismatch error can be generated if there is a mismatch in either of the quantities `start` or `length` of the dimension.

Since, Python does not use lower_bounds, we can update it to only include the `length` value in the error message. I am slightly concerned about cases when the length dimensions are equal/match but there is a mismatch in the start dimension (due to some internal bug in the compiler). In such a case the error message would just print the length dimension and this would not make it clear where exactly the mismatch occurred.

Reply:
`i32[0,4]` this seems confusing. We probably need a better approach/representation like `i32[0:4]`. I think we can look into this later. For now, I will update the code to only include the `length` dimension.

Reply:
Since this is in LPython where the lower bound is always zero, then don't print it at all. The syntax `i32[0,4]` means a 2D array, where the size of the first dimension is 0 (which I am not sure is even allowed).